### PR TITLE
Adding a sample TypeSpec ARM project created from template for Out-of-box Green CI

### PR DESCRIPTION
**Pull request by allenjzhang**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- For help with Step 2 / "ARM review", see https://aka.ms/azsdk/pr-arm-review.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Other, please clarify:
    -  Sample TypeSpec generated ARM project

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

File: specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+emit:
+  - '@azure-tools/typespec-autorest'
+options:
+  '@azure-tools/typespec-autorest':
+    use-read-only-status-schema: true
+    emitter-output-dir: "{project-root}/.."
```
Comment (weshaggard) [inline specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml:6]: Are all these properties necessary even in this basic case? Or can some of them rely on defaults?

Comment (allenjzhang) [reply specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml:6]: `use-read-only-status-schema: true`  -> Mitigates the ProvisioningStateReadOnly lint error
`emitter-output-dir: "{project-root}/.."` and  ` output-file: `  -> Need to for correct swagger file placement

**Comment by allenjzhang** [comment]
@weshaggard, @mikeharder  the [Staging Swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/pull/27502/checks?check_run_id=20913665100) error on [PatchPropertiesCorrespondToPutProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-properties-correspond-to-put-properties.md) is not correct. Looks like it did not follow the ref link to find complete properties list. I know it is staging, still it would occur for all ARM specs. Who should I file bug to get this fixed?

**Comment by konrad-jamrozik** [comment]
> @weshaggard, @mikeharder the [Staging Swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/pull/27502/checks?check_run_id=20913665100) error on [PatchPropertiesCorrespondToPutProperties](https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-properties-correspond-to-put-properties.md?rgh-link-date=2024-01-26T22%3A31%3A48Z) is not correct. Looks like it did not follow the ref link to find complete properties list. I know it is staging, still it would occur for all ARM specs. Who should I file bug to get this fixed?

@AkhilaIlla @rkmanda FYI

**Comment by allenjzhang** [comment]
@weshaggard, this is ready to merge. At least the out-of-box project has not obvious LintDiff/Model validation errors.

**Comment by allenjzhang** [comment]
@konrad-jamrozik , what is the PR Summary failure about? I did checked appropriate boxes in the description.

**Comment by weshaggard** [comment]
@allenjzhang lets please get this green so we can be sure things are working. For the ARM review requirement we should just apply the approval label to get that green.

**Comment by konrad-jamrozik** [comment]
> @konrad-jamrozik , what is the PR Summary failure about? I did checked appropriate boxes in the description.

@allenjzhang you tripped over this:

- https://github.com/Azure/azure-sdk-tools/issues/6673

I.e. automation wants for you to do this:
- https://github.com/Azure/azure-rest-api-specs/pull/27502#issuecomment-1906925946

I am attempting to circumvent it by adding label `RPaaS`. Let's see if it works.

**Comment by konrad-jamrozik** [comment]
@allenjzhang to follow-up to my previous comment [here](https://github.com/Azure/azure-rest-api-specs/pull/27502#issuecomment-1915398175), adding the `RPaaS` label doesn't unblock the automation. But you should be able to unblock it by adding `openapi-subtype: rpaas` or `openapi-subtype: providerHub` to `readme.md` as e.g. seen [here](https://github.com/Azure/azure-rest-api-specs/pull/24501/files#diff-e2dda409ecb078c959c960b7c66b416223819d005bda3f2ff278352864e57b3e):
image

File: specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/preview/2021-10-01-preview/openapi.json [hunk 2]
```diff
@@ -0,0 +1,579 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft.Contoso management service",
+    "version": "2021-10-01-preview",
```
Comment (visingla-ms) [inline specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/preview/2021-10-01-preview/openapi.json:5]: Should the API version be something recent? :)

File: specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/preview/2021-10-01-preview/openapi.json [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+  "tags": [
+    {
+      "name": "Operations"
+    },
+    {
+      "name": "Employees"
+    }
+  ],
+  "paths": {
+    "/providers/Microsoft.Contoso/operations": {
```
Comment (visingla-ms) [inline specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/preview/2021-10-01-preview/openapi.json:50]: Do we also need to include more example (maybe in subsequent PRs) about how to add various types of operations, such as Post Action on a resource, or a nested tracked/proxy resource etc.?

**Comment by visingla-ms** [comment]
@konrad-jamrozik It doesn't show me that RPSaaS check blocks this PR, is that fine?

**Comment by konrad-jamrozik** [comment]
@visingla-ms I am already working with @allenjzhang to address this. See here: https://github.com/Azure/azure-rest-api-specs/pull/27502#issuecomment-1915427553

**Comment by mentat9** [comment]
@allenjzhang, @konrad-jamrozik - Does this PR actually need review by ARM? I see @visingla-ms left a couple of unaddressed comments. If it doesn't require review, please keep it out of the queue (leave it in ARMChangesRequested state).

**Comment by konrad-jamrozik** [comment]
@mentat9 sorry for the confusion. We will bypass the ARM review. @allenjzhang to bypass it, we need to ensure that from labels with `ARM` in them, only `ARMReview` and `ARMSignedOff` labels are present.

**Pull request closed without merging**